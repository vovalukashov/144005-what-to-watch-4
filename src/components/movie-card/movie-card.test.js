import React from "react";
import renderer from "react-test-renderer";
import MovieCard from "./movie-card";

const movie = {
  name: `Сияние`,
  image: `https://birdinflight.imgix.net/wp-content/uploads/2018/06/shining-sequel-movie_cover.jpg?fm=pjpg&q=70&fit=crop&crop=faces&w=630&h=446`,
};

const handleMovieCardEnter = jest.fn();
const handleMovieTitleClick = jest.fn();

describe(`MoviesList snapshot`, () => {
  it(`MoviesList renderer`, () => {
    const tree = renderer
      .create(<MovieCard movie={movie} onMovieCardEnter={handleMovieCardEnter} onMovieTitleClick={handleMovieTitleClick} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
