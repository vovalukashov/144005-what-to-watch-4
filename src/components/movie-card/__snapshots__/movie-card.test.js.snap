// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesList snapshot MoviesList renderer 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  onClick={[Function]}
  onMouseEnter={[Function]}
>
  <div
    className="small-movie-card__image"
  >
    <img
      alt="Fantastic Beasts: The Crimes of Grindelwald"
      height="175"
      src="https://resizing.flixster.com/hfmSxiWV1FlY5vtodiys3hYXPic=/fit-in/200x296.2962962962963/v1.bTsxMTE2ODAzMjtqOzE4NTM0OzEyMDA7ODAwOzEyMDA"
      width="280"
    />
  </div>
  <h3
    className="small-movie-card__title"
  >
    <a
      className="small-movie-card__link"
      href="movie-page.html"
    >
      Сияние
    </a>
  </h3>
</article>
`;
